spec/requests/documents_spec.rb

```diff
@@ -83,11 +85,18 @@
   end
 
   describe 'GET /documents/:id' do
-    it_behaves_like 'download evidence document' do
-      subject(:show_document) { get document_path(document) }
+    subject(:show_document) { get document_path(document) }
 
+    it_behaves_like 'download evidence document' do
       let(:expected_disposition) { 'inline' }
     end
+
+    context 'when the preview does not exist' do
+      let(:document) { create :document, external_user: document_owner }
+      let(:document_owner) { external_user }
+
+      it { expect { show_document }.to raise_error(ActiveRecord::RecordNotFound) }
```

Comment:
as for the other comment regarding the raising of the error. what happens if we do not raise the error here, will it not bubble up to sentry anyway? does the view hide the show link if there is no preview anyway?

Reply:
The main reason to raise an error here is to generate a 404 response. And yes, the link does not appear if there is no preview so this is not likely to happen anyway.